tools/ClassGen/InstrBuilder.h

```diff
@@ -51,12 +66,14 @@ inline OperandKind negateOperandKind(OperandKind CC) {
     return OperandKind::In;
   case OperandKind::InOut:
     return OperandKind::InOut;
+  case OperandKind::Scratch:
+    return OperandKind::Scratch;
   }
   llvm_unreachable("Invalid operand kind.");
 }
 
 inline const char *getOperandKindStr(OperandKind CC) {
-  const char *names[] = {"In", "Out", "InOut", nullptr};
+  const char *names[] = {"In", "Out", "InOut", "Out", nullptr};
```

Comment:
Can you call it Scratch instead?

Reply:
I'm not sure I can use Scratch here because these are the actual operand kinds defined in IR for instructions. It would mean to define a new IR operand kind ... not sure we want this ...

Reply:
Yes, please try it out. It would be nice if it works! 

But I'm suspect we'd need to teach the IROptimizer and BufferSharing about this new kind of operand and touch a lot of places. In, InOut, Out are currently used in many places when we reason about liveness of buffers, etc.

Reply:
I can try but I see no reason why we shouldn't use `Out` in this case. I fear that we might run into a lot of troubles by defining a new operand kind. After all a scratch operand is an `Out` operand which has no other users.

Reply:
I tried adding a new operand kind for this called `Scratch` and I started to get errors in IR:
```
glow/lib/IR/IR.cpp:315: void verifyLiveness(const glow::IRFunction &): Assertion `(Op.second == OperandKind::Out || entry->second) && "@in and @inout operands should be initialized before their " "first use"' failed.
```
As I said I don't think we should go through this path ...

Reply:
Ok no worries, we can try to fix in the future.